src/update_handler.c

```diff
@@ -382,6 +384,20 @@ static gboolean run_slot_hook(const gchar *hook_name, const gchar *hook_cmd, Rau
 	}
 	g_subprocess_launcher_setenv(launcher, "RAUC_MOUNT_PREFIX", r_context()->config->mount_prefix, TRUE);
 
+	if (r_context()->runtime->mounted_bundle) {
+		gchar **hashes = NULL;
+
+		hashes = get_pubkey_hashes(r_context()->runtime->mounted_bundle->verified_chain);
+
+		while (*hashes != NULL) {
+			g_string_append_printf(cert_pkey_hashes, "%s", (gchar*) *hashes);
+			g_free(*hashes);
+			g_string_append_printf(cert_pkey_hashes, "%c", *++hashes ? ' ' : '\0');
+		}
```

Comment:
Use g_strjoinv() instead. By switching get_pubkey_hashes to return the g_ptr_array, you could simplify memory management here.

Reply:
`g_strjoinv()` popped from my mental stack, yes we should use it! I'm not sure about the second point, yet.